// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum BigqueryreservationAssignmentJobTypeEnum {
  BigqueryreservationAssignmentJobTypeEnumNO_VALUE_DO_NOT_USE = 0;
  BigqueryreservationAssignmentJobTypeEnumJOB_TYPE_UNSPECIFIED = 1;
  BigqueryreservationAssignmentJobTypeEnumPIPELINE = 2;
  BigqueryreservationAssignmentJobTypeEnumQUERY = 3;
}

enum BigqueryreservationAssignmentStateEnum {
  BigqueryreservationAssignmentStateEnumNO_VALUE_DO_NOT_USE = 0;
  BigqueryreservationAssignmentStateEnumSTATE_UNSPECIFIED = 1;
  BigqueryreservationAssignmentStateEnumPENDING = 2;
  BigqueryreservationAssignmentStateEnumACTIVE = 3;
}

message BigqueryreservationAssignment {
  string name = 1;
  string assignee = 2;
  BigqueryreservationAssignmentJobTypeEnum job_type = 3;
  BigqueryreservationAssignmentStateEnum state = 4;
  string project = 5;
  string location = 6;
  string reservation = 7;
}

message ApplyBigqueryreservationAssignmentRequest {
  BigqueryreservationAssignment resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteBigqueryreservationAssignmentRequest {
  string service_account_file = 1;
  BigqueryreservationAssignment resource = 2;
}

message ListBigqueryreservationAssignmentRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
  string Reservation = 4;
}

message ListBigqueryreservationAssignmentResponse {
  repeated BigqueryreservationAssignment items = 1;
}

service BigqueryreservationAssignmentService {
  rpc ApplyBigqueryreservationAssignment(ApplyBigqueryreservationAssignmentRequest) returns (BigqueryreservationAssignment);
  rpc DeleteBigqueryreservationAssignment(DeleteBigqueryreservationAssignmentRequest) returns (google.protobuf.Empty);
  rpc ListBigqueryreservationAssignment(ListBigqueryreservationAssignmentRequest) returns (ListBigqueryreservationAssignmentResponse);
}
